modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/services/Addresses.scala

```diff
@@ -61,5 +65,11 @@ object Addresses {
         unconfirmed = Balance(offChainBalance, offChainAssets.map(TokenAmount(_)))
       )) ||> trans.xa
     }
+
+    def totalBalanceOf_(address: Address): F[TotalBalance] =
+      for {
+        confirmed   <- confirmedBalanceOf(address, 0)
+        unconfirmed <- memprops.getUnconfirmedBalanceByAddress(address, confirmed)
+      } yield TotalBalance(confirmed, unconfirmed)
```

Comment:
It seems this method does the same thing as `totalBalanceOf` but in a less efficient way. `memprops.getUnconfirmedBalanceByAddress` operates on the same data from database as 
```
        offChainBalance <- uOutputRepo.sumUnspentByErgoTree(tree)
        offChainAssets  <- uAssetRepo.aggregateUnspentByErgoTree(tree)
```
The difference only is that old code aggregates balances inside DB and you does all this in the application's runtime. 
I believe this doesn't solve the problem.

Reply:
Oh,got it. The old one includes outputs from unconfirmed transactions, so all you need to do is to sum unconfirmed inputs and subtract them from total unconfirmed balance.

Reply:
Can you elaborate why simple aggregation of unconfirmed inputs/outputs doesn't work?
You get the same set of inputs/outputs inside unconfirmed transactions, so the result should be the same, right?